examples/common.cpp

```diff
@@ -92,414 +92,315 @@ void process_escapes(std::string& input) {
     input.resize(output_idx);
 }
 
+bool validate_params(const std::string& arg, int argc, int& i, char** argv, const gpt_params& default_params, bool optional = false) {
+    if (++i >= argc) {
+        if (optional) {
+            // Argument is optional and not present, return false
+            return false;
+        } else {
+            fprintf(stderr, "error: invalid parameter for argument: %s\n", arg.c_str());
+            throw std::runtime_error("Invalid parameter for argument: " + arg);
+        }
+    }
+
+    const std::string& nextArg = argv[i];
+
+    if (nextArg.empty() || (nextArg[0] == '-' && !std::isdigit(nextArg[1]))) {
```

Comment:
There is a possible out of bounds access if `nextArg` is just `"-"`.

Reply:
think this fixed it
```
if (nextArg.empty() || (nextArg.size() >= 2 && nextArg[0] == '-' && !std::isdigit(nextArg[1]))) {
```